Formula/asdf.rb

```diff
@@ -4,30 +4,24 @@ class Asdf < Formula
   url "https://github.com/asdf-vm/asdf/archive/v0.8.0.tar.gz"
   sha256 "9b667ca135c194f38d823c62cc0dc3dbe00d7a9f60caa0c06ecb3047944eadfa"
   license "MIT"
-  revision 1
+  revision 2
   head "https://github.com/asdf-vm/asdf.git"
 
   bottle :unneeded
-
-  depends_on "autoconf"
-  depends_on "automake"
   depends_on "coreutils"
-  depends_on "libtool"
-  depends_on "libyaml"
-  depends_on "openssl@1.1"
-  depends_on "readline"
-  depends_on "unixodbc"
-
-  conflicts_with "homeshick",
-    because: "asdf and homeshick both install files in lib/commands"
+  depends_on "git"
 
   def install
     bash_completion.install "completions/asdf.bash"
     fish_completion.install "completions/asdf.fish"
     zsh_completion.install "completions/_asdf"
-    libexec.install "bin/private"
-    prefix.install Dir["*"]
-    touch prefix/"asdf_updates_disabled"
+    libexec.install Dir["*"]
+    touch libexec/"asdf_updates_disabled"
+    bin.write_exec_script (opt_libexec/"bin/asdf")
```

Comment:
```suggestion
    bin.write_exec_script (opt_libexec/"bin/asdf")
    (prefix/"asdf.sh").write "source #{opt_libexec}/asdf.sh\n"
    (prefix/"asdf.fish").write "source #{opt_libexec}/asdf.fish\n"
```

Continuing discussion from main thread. This is an optional change that helps for a seamless migration so that Homebrew users that run `brew upgrade` don't have commands/shims not working. If there is a more brew-style way of doing this, then please offer other suggestions.

If something like this is not done, the alternative would be to depend on documenting changes. e.g.
- Since some Homebrew users will end up going to ASDF repo directly, ASDF team will probably need documentation changes ready to go live with PR merge. @jthegedus
- On Homebrew side, the only documentation that users may read is the caveats (if at all), but it looks like this might not be an option according to #54315. Maybe a follow up question for Homebrew team from that PR (maybe @SMillerDev ? ):
  - I did notice that some formulas like chruby do have this type of caveats:
     ```ruby
      def caveats
        <<~EOS
          Add the following to the ~/.bash_profile or ~/.zshrc file:
            source #{opt_pkgshare}/chruby.sh
     ```
  - Is there a difference in chruby's situation compared to asdf? 
     ```ruby
      def caveats
        <<~EOS
          Add the following to the ~/.bash_profile or ~/.zshrc file:
            source #{opt_libexec}/asdf.sh
     ```

Reply:
>Is there a difference in chruby's situation compared to asdf?

There isn't, but note that this caveat
```ruby
 def caveats
   <<~EOS
     Add the following to the ~/.bash_profile or ~/.zshrc file:
       source #{opt_libexec}/asdf.sh
```
is different from the one proposed in #54315.

Reply:
@carlocab thanks for the information. I added another change suggestion based on other formula caveats, so that this can be discussed separately.
- The change in this thread would help existing Homebrew+asdf users upgrade without any noticeable impact.
- The caveat addition would be best for new Homebrew+asdf users to be aware of Homebrew's location of shell scripts and hopefully avoid creating asdf GitHub issues.
- I guess the main risk of adding both would be existing Homebrew+asdf users sourcing both scripts, though the only impact would be a couple 'ms' of shell startup time.

Reply:
@cho-m Thanks for this bud, I agree with you but with small nitpicks that I don't feel strongly about, do with them what you wish: 

Nitpick 1 
Caveat on that you need to source should be optional for the feature it provides. 
This is as nit-picky as they come, but...

First off, again, sourcing the file is only necessary if you want `$ asdf shell` to work, I don't so I can live without it. In fact it's not necessary for anything else, and users can export themelves from 
`$ asdf export-shell-version shell <language> <version>.` I didn't notice it was gone until you told me, my work flow so far has been fine with switching between global and local installations. 


Nitpick 2
Even more extreme nit-picky, while we're on a roll.
I read somewhere that `source` might or was going to be banned from `asdf` source code  because it's not posix compliant, not sure if it matters on macOS and again, I am not going to use it, but could switch that to `.` instead. 

Apart from that, looks great and I appreciate you taking the time to test, that's something I wanted and no one else did until now, much obliged!

Reply:
@cho-m 

I think I found a mistake.

But your change writes so they source
```ruby    
(prefix/"asdf.sh").write "source #{opt_libexec}/asdf.sh\n"
```

Would not source the correct file, you might want something like 

```ruby    
(prefix/"asdf.sh").write "source #{opt_libexec}/lib/asdf.sh\n"
```

But I am wondering if that will cause issues. 

I mean
`asdf/asdf.sh` and `asdf/asdf.fish` already sources the file in question 

It has this at the bottom. 
```
# shellcheck source=lib/asdf.sh
# Load the asdf wrapper function
. "${ASDF_DIR}/lib/asdf.sh"
```

Where `${ASDF_DIR}` is supposed to be (not saying it is or it works) `"$(dirname "$current_script_path")"`


Edit: Ops accidentally resolved. Sorry about that.

Reply:
@cho-m Sorry, I think I misunderstood what you wanted to do with my previous comment. 
I see now you wanted to make sure current paths work fine for existing users. 
My bad! 
I added your changes.